LookUpGroup27/LookUp
# feat: enforce unique usernames during registration and enhance related UI and tests

This pull request introduces distinct username validation into the user registration flow, ensuring that no two users can share the same username.

**Key Changes:**
- **New Feature:**
  - Introduce `UsernameAlreadyExistsException` to handle scenarios where the chosen username is already taken.
  - Update `RegisterRepository` and `RegisterRepositoryFirestore` to query Firestore for existing usernames before creating a user.
  - Extend `RegisterState` and `RegisterViewModel` to support username fields and validation logic.
  - Enhance the registration UI (`Register.kt`) with a username input field and proper error handling.

- **Tests:**
  - Add and refine tests in `RegisterViewModelTest` and `RegisterRepositoryFirestoreTest` to cover username uniqueness checks and related exceptions.
  - Update UI tests (`RegisterKtTest`) to verify the presence and correct behavior of the username field.

By merging this PR, we ensure that the registration process provides immediate feedback on username availability, improving both user experience and data integrity.
